client/e2e-test/cypress/integration/index.spec.js

```diff
@@ -124,290 +53,237 @@ context('Akvo Lumen', () => {
-            cy.get('.saveStatus').should('contain', 'All changes saved');
-            // chart elements exist
-            cy.get('.Chart path').should('exist');
-          })
-
-          .then(() => {
-            // Pivot
-            cy.get('[data-test-id="visualisation-type-pivot-table"]').click({
-              force: true,
-            });
-
-            // wait for changes to be saved
-            cy.get('.saveStatus').should('contain', 'All changes saved');
-            // chart elements exist
-            cy.get('.PivotTable th')
-              .contains('Bob')
-              .should('exist');
-            cy.get('.PivotTable th')
-              .contains('Jane')
-              .should('exist');
-            cy.get('.PivotTable th')
-              .contains('Frank')
-              .should('exist');
-            cy.get('.PivotTable th')
-              .contains('Lisa')
-              .should('exist');
-            cy.get('.PivotTable td')
-              .contains('Total')
-              .should('exist');
-            cy.get('.PivotTable td.cell')
-              .contains('1')
-              .should('exist');
-            cy.get('.PivotTable .categoryColumnTitle')
-              .contains(COLUMNS.TEXT_1)
-              .should('exist');
-          })
-
-          .then(() => {
-            // back to library
-            cy.get('[data-test-id="back-button"]').click();
-          });
+        .type(`${datasetName}{enter}`, { force: true });
+
+      // open column menu
+      cy.get('[data-test-id="column-select"] .SelectMenu input', {
+        timeout: 10000,
+      })
+        .first()
+        .click({ force: true })
+        .type(`${COLUMNS.TEXT_1}{enter}`, { force: true });
+
+      const title = `Visualisation of ${datasetName}`;
+
+      // Focusing title
+      cy.get('div[data-test-id="entity-title"]').click();
+
+      // Typing visualisation name
+      cy.get('input[data-test-id="entity-title"]').type(title);
+
+      // Saving visualisation
+      cy.get('body').click();
+
+      // wait for changes to be saved
+      cy.get('.saveStatus').should('contain', 'All changes saved');
+
+      // pivot table elements exist
+      cy.get('.PivotTable .title').contains(title).should('exist');
+      cy.get('.PivotTable th').contains('Bob').should('exist');
+      cy.get('.PivotTable th').contains('Jane').should('exist');
+      cy.get('.PivotTable th').contains('Frank').should('exist');
+      cy.get('.PivotTable th').contains('Lisa').should('exist');
+      cy.get('.PivotTable td').contains('Total').should('exist');
+      cy.get('.PivotTable td.cell').contains('1').should('exist');
+      cy.get('.PivotTable .categoryColumnTitle')
+        .contains(COLUMNS.TEXT_1)
+        .should('exist');
+
+      // back to library
+      cy.get('[data-test-id="back-button"]').click();
+    });
+
+    context('create visualisation from dataset page', () => {
+      cy.get('.dataset').contains(datasetName).click({ force: true });
```

Comment:
* When searching for a DOM element we should rely on a *stable* identifier. That's why we have added some many `data-test-id=<something>` - 
* Doing a search based just a `className` feels *brittle*

Reply:
If you're suggesting we use `data-test-id` for all elements used in the test suite, I can do that in a separate PR. Because that's a lot of changes